### PR TITLE
Pass parses filename as parameter

### PR DESCRIPTION
**Pull request by glicerico**
Syncing from singnet.
Changes are related to passing filename for mst-parses output as parameter, in order do avoid mixing parses when processing in parallel (preventing for socket use).

### CODE REVIEW

File: tests/MST-parse-test.scm [hunk 1]
```diff
@@ -65,8 +65,8 @@
 )
 
 ; Parse the sentences
-(define parse-1 (observe-mst-mode test-str-1 cnt-mode dist-mult #f))
-(define parse-2 (observe-mst-mode test-str-2 cnt-mode dist-mult #f))
+(define parse-1 (observe-mst-mode test-str-1 cnt-mode dist-mult "#f"))
+(define parse-2 (observe-mst-mode test-str-2 cnt-mode dist-mult "#f"))
```
Comment (linas) [inline tests/MST-parse-test.scm]: Ughh. The quoted string `#f` ? That's going to lead to confusion. If you want to have a string here, change it to `"empty"` or `"none"` or something that doen't collide with a reserved keyword.

File: run-poc/redefine-mst-parser.scm [hunk 2]
```diff
@@ -230,17 +230,16 @@
 		(lambda (dj) (if (not (is-oversize? dj)) (count-one-atom dj)))
 		(make-sections parse)
 	)
-	(if EXPORT-MST
+	(if (not (equal? EXPORT-MST "#f"))
```
Comment (linas) [inline run-poc/redefine-mst-parser.scm]: OH, I see the problem.  This one change is forcing you to use `"#f"` I think the original form was correct:  either `EXPORT-MST` is false, or `EXPORT-MST` is a string holding a filename.  Maybe you should keep the old version, and just change `EXPORT-MST` to `MST-FILENAME`

**Comment by linas** [comment]
I can merge this, but maybe you want to revisit these changes? They don't quite look well-designed.  Let me know.

**Comment by linas** [comment]
Also: can you set up a pull reuest in the opposite direction? The singnet repo is many months and many hundreds of commits behind the opencog repo. They should be kept in sync.

File: run-poc/redefine-mst-parser.scm (hunk 2 above)
Comment (glicerico) [reply run-poc/redefine-mst-parser.scm]: The problem with the original approach is that I need to pass through netcat a variable that is either a string or a bool, so I need to pass it using 
> 			print NC "($ARGV[2] \"$buffer\" \"$ARGV[3]\" '(@dist_mult) \"$ARGV[5]\")\n";

I think I will use your suggestion of using "none" as a keyword for no exporting.

**Comment by glicerico** [comment]
@linas , changed the keyword to avoid confusion with #f

**Pull request merged**